src/main/scala/com/ppb/rabbitconsumer/RabbitConsumer.scala

```diff
@@ -11,6 +12,9 @@ import scalaz.concurrent.Task
 import scalaz.stream._
 
 case class Cxn(filename: String, nextMessage: () => RabbitResponse, disconnect: () => Try[Unit])
+case class Sxn(subscription:((Try[RabbitResponse] => Unit) => String), disconnect: () => Try[Unit])
```

Comment:
Let's have a chat about how we could refactor `Sxn`.  It looks like it's a case class that holds functions that manage a user's attempt to subscribe to a Rabbit queue.  Perhaps it doesn't belong in a case class and we could capture some logic in a plan old `object`.